compiler/hash-token/src/lib.rs

```diff
@@ -272,6 +280,46 @@ impl std::fmt::Display for TokenKind {
     }
 }
 
+/// Represents the featured base for numeric literals.
+#[derive(Debug)]
+pub enum Base {
+    /// Binary base, denoted in literals as `0b101010`
+    Binary,
+    /// Octal base, denoted in literals as `0o26317261`
+    Octal,
+    /// Decimal base, written as `102391`
+    Decimal,
+    /// Hexadecimal base, written as `0xdeadbeef`
+    Hex,
+    /// Unsupported base, the language doesn't support the
+    /// provided radix as a base.
+    Unsupported,
```

Comment:
Why is this needed?

Reply:
It's not really needed, it's just that I thought that maybe instead of using an `unwrap()` when converting the `u32` into `Base`, it wouldn't panic and just fails gracefully. By the term "gracefully", I mean that it would say "unsupported base", and then someone can submit a bug report if that ever does occur which it shouldn't.

Reply:
Maybe this should live inside lexer then